node_modules/oae-core/topnavigation/js/topnavigation.js

```diff
@@ -136,7 +136,7 @@ define(['jquery', 'oae.core', 'activityadapter'], function($, oae, ActivityAdapt
                 // full control of the URL hash and adding one  into the URL ourself would interfere with that
                 // @see https://github.com/oaeproject/3akai-ux/issues/3872
                 query = query.replace(/#/g, '');
-                window.location = '/search/' + oae.api.util.security().encodeForURL(query);
+                window.location = '/search/all?q=' + oae.api.util.security().encodeForURL(query);
```

Comment:
For logged in users, searching from the top navigation can point to the "My Stuff" page by default.